src/math/unary_ops.ts

```diff
@@ -121,8 +122,14 @@ export class Ops {
         (min <= max),
         `Error in clip: min (${min}) must be` +
             `less than or equal to max (${max}).`);
-    return ENV.engine.executeKernel('Clip', {inputs: {x}, args: {min, max}}) as
-        T;
+    return ENV.engine.executeKernel(
+        'Clip', {inputs: {x}, args: {min, max}}, (dy: T, y: T) => {
+      return {
+          x: () => dy.where(
```

Comment:
Gradient for `clip` is not defined for x = min and x = max. We should add a TODO here to fix this once it is defined. Reference: #606

Reply:
Done.